test/acceptance/test/profiles_test.go

```diff
@@ -129,6 +129,29 @@ Namespace: %s`, clusterName, namespace)))
 			resp, statusCode, err = kubernetesDoRequest(namespace, clusterName+"-"+profileName, "9898", "/healthz", clientSet)
 			return statusCode
 		}, "120s", "1s").Should(Equal(http.StatusOK))
+
+		By("Updating the version of the installed profile")
+		time.Sleep(4 * time.Minute)
+		Eventually(func() string {
+			stdOut, stdErr = runCommandAndReturnStringOutput(fmt.Sprintf("%s update profile --name %s --version 6.0.0 --namespace %s --cluster %s --config-repo %s --auto-merge", gitopsBinaryPath, profileName, namespace, clusterName, appRepoRemoteURL))
+			return stdErr
+		}, "120s", "5s").Should(BeEmpty())
```

Comment:
isn't this going to re-run `update profile` every 5s?

Reply:
Yep so due to the Github API issue, it'll fail until the `GET` request returns the contents of `profiles.yaml` with:
```
Error: failed to find installed profiles in '.weave-gitops/clusters/kind-kind/system/profiles.yaml' of config repo "ssh://git@github.com/nikimanoledaki/fun-times.git"
```

Pinging until it returns the changes would be okay if it doesn't count towards the rate limit. To do this, we could make conditional requests that check if anything has changed like [this](https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#conditional-requests). If the response's status code is `304` (meaning that nothing has changed), Github kindly does not count that request towards the rate limit.

That being said, I don't think we need to worry about this here.